pure_pursuit_wrapper/src/pure_pursuit_wrapper_node.cpp

```diff
@@ -34,8 +34,11 @@ int main(int argc, char** argv) {
   message_filters::Synchronizer<pure_pursuit_wrapper::PurePursuitWrapper::SyncPolicy> sync(pure_pursuit_wrapper::PurePursuitWrapper::SyncPolicy(100), PurePursuitWrapper.pose_sub, PurePursuitWrapper.trajectory_plan_sub);
   sync.registerCallback(boost::bind(&pure_pursuit_wrapper::PurePursuitWrapper::TrajectoryPlanPoseHandler, &PurePursuitWrapper, _1, _2));
 
+  ros::Rate r(10);
   while (ros::ok() && !PurePursuitWrapper.shutting_down_) {
-    ros::CARMANodeHandle::spin();
+    PurePursuitWrapper.PublishPluginDiscovery();
+    ros::spinOnce();
+    r.sleep();
   }
```

Comment:
Looks like in the previous version there was a misuse of CARMANodeHandle::spint(). Rather than updating the loop here. The loop should be replaced with the appropriate spin callback and rate setting. Otherwise PublishPluginDiscovery will not be exception handled.